diff -ur ctpp2-2.8.3/CMakeLists.txt ctpp2-2.8.3.patched/CMakeLists.txt
--- ctpp2-2.8.3/CMakeLists.txt	2017-01-17 10:09:06.259494234 +0100
+++ ctpp2-2.8.3.patched/CMakeLists.txt	2017-01-17 11:48:18.427522886 +0100
@@ -785,7 +785,7 @@
 
     INSTALL(TARGETS ctpp2-static DESTINATION lib${LIB_SUFFIX})
 
-    INSTALL(TARGETS ctpp2 DESTINATION .)
+    INSTALL(TARGETS ctpp2 DESTINATION lib${LIB_SUFFIX})
 
     FILE(WRITE ctpp2-config.cmd ${CTPP_COMPILER_HELPER})
 
