diff -ur ctpp2-2.8.3/src/CTPP2StringIconvOutputCollector.cpp ctpp2-2.8.3.iconv/src/CTPP2StringIconvOutputCollector.cpp
--- ctpp2-2.8.3/src/CTPP2StringIconvOutputCollector.cpp	2017-02-07 10:42:40.567806420 +0100
+++ ctpp2-2.8.3.iconv/src/CTPP2StringIconvOutputCollector.cpp	2017-01-30 15:22:19.734186564 +0100
@@ -38,6 +38,8 @@
 namespace CTPP // C++ Template Engine
 {
 
+
+#ifdef ICONV_SUPPORT
 //
 // Constructor
 //
@@ -114,5 +116,7 @@
 return 0;
 }
 
+#endif //ICONV_SUPPORT
+
 } // namespace CTPP
 // End.
