diff -ur ctpp2-2.8.3/include/CTPP2VMExecutable.hpp ctpp2-2.8.3.patched/include/CTPP2VMExecutable.hpp
--- ctpp2-2.8.3/include/CTPP2VMExecutable.hpp	2012-08-02 09:22:44.000000000 +0200
+++ ctpp2-2.8.3.patched/include/CTPP2VMExecutable.hpp	2017-01-17 11:47:08.332691919 +0100
@@ -52,7 +52,7 @@
   @struct VMExecutable CTPP2VMExecutable.hpp <CTPP2VMExecutable.hpp>
   @brief CTPP Executable file
 */
-struct VMExecutable
+struct CTPP2DECL VMExecutable
 {
 	/** CTPP magic number                    */
 	UCHAR_8      magic[4]; // 'CTPP' xor 0xFFFFFFFF
