diff -ur xapian-core-1.4.0/bin/Makefile.mk xapian-core-1.4.0.no_lz/bin/Makefile.mk
--- xapian-core-1.4.0/bin/Makefile.mk	2016-06-25 17:36:49.000000000 +0200
+++ xapian-core-1.4.0.no_lz/bin/Makefile.mk	2017-02-07 11:15:10.166507000 +0100
@@ -66,7 +66,7 @@
 	unicode/unicode-data.cc\
 	unicode/utf8itor.cc
 
-bin_xapian_inspect_LDADD = $(ldflags) libgetopt.la -lz
+bin_xapian_inspect_LDADD = $(ldflags) libgetopt.la $(libxapian_la)
 
 bin_xapian_progsrv_SOURCES = bin/xapian-progsrv.cc
 bin_xapian_progsrv_LDADD = $(ldflags) libgetopt.la $(libxapian_la)
diff -ur xapian-core-1.4.0/Makefile.in xapian-core-1.4.0.no_lz/Makefile.in
--- xapian-core-1.4.0/Makefile.in	2016-06-25 17:39:28.000000000 +0200
+++ xapian-core-1.4.0.no_lz/Makefile.in	2017-02-07 11:27:21.050507000 +0100
@@ -707,7 +707,8 @@
 	unicode/bin_xapian_inspect-unicode-data.$(OBJEXT) \
 	unicode/bin_xapian_inspect-utf8itor.$(OBJEXT)
 bin_xapian_inspect_OBJECTS = $(am_bin_xapian_inspect_OBJECTS)
-bin_xapian_inspect_DEPENDENCIES = $(am__DEPENDENCIES_1) libgetopt.la
+bin_xapian_inspect_DEPENDENCIES = $(am__DEPENDENCIES_1) libgetopt.la \
+	$(libxapian_la)
 am_bin_xapian_progsrv_OBJECTS = bin/xapian-progsrv.$(OBJEXT)
 bin_xapian_progsrv_OBJECTS = $(am_bin_xapian_progsrv_OBJECTS)
 bin_xapian_progsrv_DEPENDENCIES = $(am__DEPENDENCIES_1) libgetopt.la \
@@ -1405,7 +1406,7 @@
 	unicode/unicode-data.cc\
 	unicode/utf8itor.cc
 
-bin_xapian_inspect_LDADD = $(ldflags) libgetopt.la -lz
+bin_xapian_inspect_LDADD = $(ldflags) libgetopt.la $(libxapian_la)
 bin_xapian_progsrv_SOURCES = bin/xapian-progsrv.cc
 bin_xapian_progsrv_LDADD = $(ldflags) libgetopt.la $(libxapian_la)
 bin_xapian_replicate_SOURCES = bin/xapian-replicate.cc
