diff -ru ctpp2-2.8.3.orig/CMakeLists.txt ctpp2-2.8.3/CMakeLists.txt
--- ctpp2-2.8.3.orig/CMakeLists.txt	2012-11-10 21:45:57.000000000 +0100
+++ ctpp2-2.8.3/CMakeLists.txt	2017-01-09 13:40:09.952086862 +0100
@@ -343,7 +343,8 @@
 # ################################################################################
 # Sources
 #
-CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/CTPP2SysHeaders.h.cmake ${CMAKE_CURRENT_SOURCE_DIR}/include/CTPP2SysHeaders.h)
+CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/CTPP2SysHeaders.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/CTPP2SysHeaders.h)
+include_directories("${CMAKE_CURRENT_BINARY_DIR}")
 #
 SET(LIBSRCS
             src/CDT.cpp
@@ -704,7 +705,7 @@
               include/CTPP2StringOutputCollector.hpp
               include/CTPP2SymbolTable.hpp
               include/CTPP2Syntax.h
-              include/CTPP2SysHeaders.h
+              ${CMAKE_CURRENT_BINARY_DIR}/CTPP2SysHeaders.h
               include/CTPP2SysTypes.h
               include/CTPP2SyscallFactory.hpp
               include/CTPP2Types.h
@@ -899,14 +900,14 @@
 ")
 
     # Write file
-    FILE(WRITE ctpp2-config ${CTPP_COMPILER_HELPER})
+    FILE(WRITE ${CMAKE_CURRENT_BINARY_DIR}/ctpp2-config ${CTPP_COMPILER_HELPER})
 
     # Install binaries
     INSTALL(FILES ${CMAKE_CURRENT_BINARY_DIR}/ctpp2vm
                   ${CMAKE_CURRENT_BINARY_DIR}/ctpp2c
                   ${CMAKE_CURRENT_BINARY_DIR}/ctpp2i
                   ${CMAKE_CURRENT_BINARY_DIR}/ctpp2json
-                  ctpp2-config
+                  ${CMAKE_CURRENT_BINARY_DIR}/ctpp2-config
             DESTINATION bin
             PERMISSIONS OWNER_EXECUTE OWNER_READ OWNER_WRITE
                         GROUP_EXECUTE GROUP_READ
Seulement dans ctpp2-2.8.3.orig/include: CTPP2SysHeaders.h
